tests/interlocking_component/conftest.py

```diff
@@ -326,3 +327,9 @@ def add_train(identifier, routeID=None, typeID=None):
 
 
 # pylint: enable=invalid-name
+
+
+@pytest.fixture
+def router() -> Router:
+    router = Router()
+    return router
```

Comment:
```suggestion
@pytest.fixture
def router() -> Router:
    return Router()
```

I am not entirely sure how python handles it. But I think it is possible that you overwrite the fixture with `router = ...`.